lib/git_reflow.rb

```diff
@@ -118,42 +123,34 @@ def deliver(options = {})
         say "No pull request exists for #{remote_user}:#{current_branch}\nPlease submit your branch for review first with \`git reflow review\`", :deliver_halted
       else
 
-        commit_message = if "#{existing_pull_request.description}".length > 0
-                           existing_pull_request.description
-                         else
-                           "#{get_first_commit_message}"
-                         end
-
-        if existing_pull_request.good_to_merge?(force: options['skip_lgtm'])
-          puts "Merging pull request ##{existing_pull_request.number}: '#{existing_pull_request.title}', from '#{existing_pull_request.feature_branch_name}' into '#{existing_pull_request.base_branch_name}'"
-
-          update_destination(base_branch)
-          merge_feature_branch(feature_branch,
-                               :destination_branch  => base_branch,
-                               :pull_request_number => existing_pull_request.number,
-                               :lgtm_authors        => existing_pull_request.approvals,
-                               :message             => commit_message)
-          committed = run_command_with_label 'git commit', with_system: true
-
-          if committed
-            say "Merge complete!", :success
-
-            # check if user always wants to push and cleanup, otherwise ask
-            always_deploy_and_cleanup = GitReflow::Config.get('reflow.always-deploy-and-cleanup') == "true"
-            deploy_and_cleanup = always_deploy_and_cleanup || (ask "Would you like to push this branch to your remote repo and cleanup your feature branch? ") =~ /^y/i
-
-            if deploy_and_cleanup
-              run_command_with_label "git push origin #{base_branch}"
-              run_command_with_label "git push origin :#{feature_branch}"
-              run_command_with_label "git branch -D #{feature_branch}"
-              puts "Nice job buddy."
-            else
-              puts "Cleanup halted.  Local changes were not pushed to remote repo.".colorize(:red)
-              puts "To reset and go back to your branch run \`git reset --hard origin/#{base_branch} && git checkout #{feature_branch}\`"
-            end
+        if existing_pull_request.good_to_merge?(force: options[:skip_lgtm])
+          # displays current status and prompts user for confirmation
+          self.status base_branch
+          
+          if self.deliver?
+            begin
+              committed = existing_pull_request.merge!(options)
+
+              if committed
+                say "Pull Request successfully merged.", :success
+
+                if self.cleanup_feature_branch?
+                  run_command_with_label "git push origin :#{feature_branch}"
+                  run_command_with_label "git branch -D #{feature_branch}"
+                  puts "Nice job buddy."
```

Comment:
We have a common practice of "refactoring in context".  Basically, since we're already modifying this code, it would make sense to refactor as long as the refactor stays within the context of the current scope.